pkg/clientgen/javascript.go

```diff
@@ -489,12 +522,23 @@ func (js *javascript) rpcCallSite(w *indentWriter, rpc *meta.RPC, rpcPath string
 	w.WriteString("\n//Populate the return object from the JSON body and received headers\nconst rtn = await resp.json()\n")
 
 	for _, headerField := range respEnc.HeaderParameters {
+		isSetCookie := strings.ToLower(headerField.WireFormat) == "set-cookie"
+		if isSetCookie {
+			w.WriteString("// Skip set-cookie header in browser context as browsers doesn't have access to read it\n")
+			w.WriteString("if (!BROWSER) {\n")
+			w = w.Indent()
```

Comment:
maybe it would be better to not have the set-cookie fields at all in the type? and then add it back when we create a nodejs server target for the clientgen?

Reply:
will address this when adding cli flags for generating for browser or server target